lib/src/source_visitor.dart

```diff
@@ -2252,7 +2254,22 @@ class SourceVisitor extends ThrowingAstVisitor {
 
   @override
   void visitNamedType(NamedType node) {
-    visit(node.name);
+    final importPrefix = node.importPrefix;
+    if (importPrefix != null) {
+      // TODO(scheglov) This is wrong, we should not create nodes here.
+      // But I don't see how to change CallChainVisitor() to work with tokens.
+      // So, I will leave it to the maintainers to figure this out.
+      visit(
+        src_ast.PrefixedIdentifierImpl(
+          prefix: src_ast.SimpleIdentifierImpl(importPrefix.name),
+          period: importPrefix.period,
+          identifier: src_ast.SimpleIdentifierImpl(node.name2),
+        ),
+      );
+    } else {
+      token(node.name2);
+    }
```

Comment:
I have a fix for this on the flutter_experiment branch that you can use. It's:

```dart
  void visitNamedType(NamedType node) {
    var importPrefix = node.importPrefix;
    if (importPrefix != null) {
      builder.startSpan();

      token(importPrefix.name);
      soloZeroSplit();
      token(importPrefix.period);
    }

    token(node.name2);
    visit(node.typeArguments);
    token(node.question);

    if (importPrefix != null) {
      builder.endSpan();
    }
  }
```

Reply:
Ah, this is much better, thank you!